src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -1044,7 +1044,14 @@ def __init__(self, *args, **kwargs):
         cert['csr_path'] = os.path.join(
             root_path, f'{cert["name"]}.csr'
         )
-
+        x509 = crypto.load_certificate(
+            crypto.FILETYPE_PEM,
+            cert['certificate']
+        )
+        cert['subject_name_hash'] = '%x' % x509.subject_name_hash()
+        cert['hash_symlink_path'] = os.path.join(
```

Comment:
subject name hash should be added where we load the cert in cryptokey service as if it fails to load etc, it will be handled appropriately in certificate extend method.